### DISCUSSION TITLE
how to disable configured nvim-dap-ui triggers/listeners

### DISCUSSION THREAD

**Discussion by 3ynm** [LazyVim/LazyVim#1458] [Q&A] [answered]
I use the REPL way more than the UI when debugging with DAP.

I'd like to disable the triggers added on `plugins.extras.dap.core` on `mfussenegger/nvim-dap -> dependencies -> rcarriga/nvim-dap-ui -> config` but haven't found a way to do this.

How can I do it?

**Comment by dpetka2001** [comment]
You could overwrite the `config` function with your own in your own configuration.

**Reply by 3ynm** [reply]
how? I tried to do it with:

```
return {
  "mfussenegger/nvim-dap",
  dependencies = {
    {
      "rcarriga/nvim-dap-ui",
      config = nil,
    },
  },
}
```

But it didn't work.

**Reply by dpetka2001** [reply, marked as answer]
Something like
```
return {
  "mfussenegger/nvim-dap",
  dependencies = {
    {
      "rcarriga/nvim-dap-ui",
	   config = function(_, opts)
         local dap = require("dap")
         local dapui = require("dapui")
         dapui.setup(opts)
	   end, 
    },
  },
}
```
I've not tested that out myself, but this is the way you overwrite the default `config` function. Obviously, you should set up the necessary `require`s yourself in order for `nvim-dap-ui` to work.

Edit: the `local dap = require("dap")` might not even be needed in this situation, since you're getting rid of the listeners.
